terkin/datalogger.py

```diff
@@ -24,25 +24,43 @@ class TerkinDatalogger:
     name = 'Terkin MicroPython Datalogger'
     version = __version__
 
+    __instance = None
+
+    @staticmethod
+    def getInstance(settings = None):
+        """ Static access method. """
+        if TerkinDatalogger.__instance == None:
+            if settings == None:
+                raise Exception("Settings are None but instance wasn't created before.")
+            else:
+                TerkinDatalogger(settings)
+        return TerkinDatalogger.__instance
+
     def __init__(self, settings):
+        if TerkinDatalogger.__instance != None:
+            raise Exception("This class is a singleton!")
+        else:
```

Comment:
No need to `else:` this block here if the `if:` will raise an exception already. So, any indentation will become unnecessary.